packages/core/plugin/src/withSentryAndroid.ts

```diff
@@ -49,3 +56,58 @@ export function modifyAppBuildGradle(buildGradle: string): string {
 
   return buildGradle.replace(pattern, match => `${applyFrom}\n\n${match}`);
 }
+
+export function modifyMainApplication(config: ExpoConfig): ExpoConfig {
+  return withMainApplication(config, async config => {
+    if (!config.modResults || !config.modResults.path) {
+      warnOnce('Skipping MainApplication modification because the file does not exist.');
```

Comment:
Let's try to be more explicit about what this means for the user -> the native Android init won't be added.

````suggestion
      warnOnce("Can't add `RNSentrySDK.init` to Android MainApplication, because the file was not found.");
````